ui/admin-portal/src/app/components/candidates/view/exam/create/create-candidate-exam.component.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2024 Talent Beyond Boundaries.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {Component, OnInit} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from "@angular/forms";
+import {NgbActiveModal} from "@ng-bootstrap/ng-bootstrap";
+import {CandidateExam, Exam} from "../../../../../model/candidate";
+import {CandidateExamService, CreateCandidateExamRequest} from "../../../../../services/candidate-exam.service";
+import {EnumOption, enumOptions} from "../../../../../util/enum";
+import {generateYearArray} from "../../../../../util/year-helper";
+
+@Component({
+  selector: 'app-create-candidate-exam',
+  templateUrl: './create-candidate-exam.component.html',
+  styleUrls: ['./create-candidate-exam.component.scss']
+})
+export class CreateCandidateExamComponent implements OnInit {
+  candidateCertification: CandidateExam;
+  candidateForm: FormGroup;
+
+  candidateId: number;
+  countries = [];
+  years = [];
+  error;
+  loading: boolean;
+  saving: boolean;
+  typing: boolean;
+
+  examOptions: EnumOption[] = enumOptions(Exam);
+
+  constructor(private activeModal: NgbActiveModal,
+              private fb: FormBuilder,
+              private candidateExamService: CandidateExamService) {
+  }
+  ngOnInit() {
+    this.loading = true;
+
+    this.years = generateYearArray(1950, true);
+
+    this.candidateForm = this.fb.group({
+      exam: ['', [Validators.required]],
+      otherExam: ['', null],
+      score: ['', [Validators.required]],
+      year: ['', [Validators.required]],
+      notes: ['', [Validators.required]]
```

Comment:
Should notes be required? I don't think that they were on the candidate portal

Reply:
Done